src/ZoneServer/World/Actors/Monsters/Mob.cs

```diff
@@ -381,47 +394,132 @@ private void DropItems(Character killer)
+			if (this.IsBuffActive(BuffId.EliteMonsterBuff)) iterations = worldConf.EliteRolls;
+			if (this.IsBuffActive(BuffId.SuperDrop)) iterations = worldConf.SilverJackpotRolls;
+			if (this.IsBuffActive(BuffId.TwinkleBuff)) iterations = worldConf.GoldJackpotRolls;
 
-				if (!ZoneServer.Instance.Data.ItemDb.TryFind(dropItemData.ItemId, out var itemData))
+			for (int i = 0; i < iterations; i++)
+			{
+				foreach (var dropItemData in this.Data.Drops)
 				{
-					Log.Warning("Monster.Kill: Drop item '{0}' not found.", dropItemData.ItemId);
-					continue;
+					var dropChance = GetAdjustedDropRate(dropItemData);
+
+					var dropSuccess = rnd.NextDouble() < dropChance / 100f;
+					if (!dropSuccess)
+						continue;
+
+					if (!ZoneServer.Instance.Data.ItemDb.TryFind(dropItemData.ItemId, out var itemData))
+					{
+						Log.Warning("Monster.Kill: Drop item '{0}' not found.", dropItemData.ItemId);
+						continue;
+					}
+
+					var dropItem = new Item(itemData.Id);
+					var minAmount = dropItemData.MinAmount;
+					var maxAmount = dropItemData.MaxAmount;
+
+					if (dropItemData.ItemId == ItemId.Silver || dropItemData.ItemId == ItemId.Gold)
+					{
+						minAmount = Math.Max(1, (int)(minAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
+						maxAmount = Math.Max(minAmount, (int)(maxAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
+					}
+
+					dropItem.Amount = rnd.Next(minAmount, maxAmount + 1);
+
+					if (killer == null || dropChance > autoloot)
+					{
+						var direction = new Direction(rnd.Next(0, 360));
+						var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
+						var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+
+						dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
+						dropItem.Drop(this.Map, this.Position, direction, distance);
+					}
+					else
+					{
+						killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
+					}
 				}
+			}
+		}
 
-				var dropItem = new Item(itemData.Id);
-				var minAmount = dropItemData.MinAmount;
-				var maxAmount = dropItemData.MaxAmount;
+		/// <summary>
+		/// Rolls for a chance for this monster to become a rare (jackpot or elite) monster
+		/// </summary>
+		/// <param name="redOrb">true if this was spawned by a red orb and should get the boosted rates</param>
+		/// <returns></returns>
+		public void PossiblyBecomeRare(bool redOrb = false)
+		{
+			var rnd = RandomProvider.Get();
 
-				if (dropItemData.ItemId == ItemId.Silver || dropItemData.ItemId == ItemId.Gold)
-				{
-					minAmount = Math.Max(1, (int)(minAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-					maxAmount = Math.Max(minAmount, (int)(maxAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-				}
+			var worldConf = ZoneServer.Instance.Conf.World;
 
-				dropItem.Amount = rnd.Next(minAmount, maxAmount + 1);
+			var spawnRateBoost = 1F;
```

Comment:
Not a big issue, but we're using lower case `f`s for floats everywhere else, and for consistency we should stick with that.

Reply:
I fixed a bunch, hopefully I didn't miss any.